pkg/controller/poolcoordinator/podbinding/podbinding_controller.go

```diff
@@ -140,7 +149,7 @@ func (r *ReconcilePodBinding) configureTolerationForPod(pod *corev1.Pod, tolerat
 
 	if toleratesNodeNotReady || toleratesNodeUnreachable {
 		klog.V(4).Infof("pod(%s/%s) => toleratesNodeNotReady=%v, toleratesNodeUnreachable=%v, tolerationSeconds=%d", pod.Namespace, pod.Name, toleratesNodeNotReady, toleratesNodeUnreachable, *tolerationSeconds)
-		_, err := r.podBindingClient.CoreV1().Pods(pod.Namespace).Update(context.TODO(), pod, metav1.UpdateOptions{})
+		err := r.Status().Update(context.TODO(), pod)
```

Comment:
It seems that `r.Status()` can only update pod status, but we need to update pod spec.